### PR TITLE
V0.38.0 Updated HTTPQL, both Automate pages, added a general Plugins …

### PR DESCRIPTION
**Pull request by ninjeeter**
…tab page in References, included index and sidebar listings.

### CODE REVIEW

**Comment by ninjeeter** [comment]
Wrote all the errors we discussed about, realized it would cause a merge conflict with my previous branch, copied what I wrote, reverted the changes, pasted into a .txt file for later anddd DIDN'T SAVE THE TXT FILE. :')

It's all ready to go though. Will expand upon some topics, such as regex and the new Plugins page once I do some background reading.

File: src/concepts/essentials/httpql.md [hunk 1]
```diff
@@ -2,99 +2,167 @@
...
 ### resp
 
-- **code**: Status code of the reponse. If the response is malformed, this will contain everything after `HTTP/1.1` and the following space.
+- `code`: The status code of the reponse. If the response is malformed, this will contain everything after `HTTP/1.1` and the following whitespace.
+- `raw`: The full raw data of the response. This allows you to search by elements that Caido currently does not index (_such as headers_).
+- `roundtrip`: The total time taken for the request to travel from the client to the server and for the response to travel back from the server to the client.
 
-- **raw**: The full raw data of the response. This allows you to search on things we currently don't index (like headers).
+::: info
+View [roundtrip values](#roundtrip) for supported time formats.
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:61]: Roundtrip is always an integer in milliseconds, the time formats are for the `created_at` field.

File: src/concepts/essentials/httpql.md [hunk 2]
```diff
@@ -2,99 +2,167 @@
...
+
+::: info
+Not all regex features are currently supported by Caido (_such as look-ahead expressions_) as they are not included in the regex library of Rust.
+:::
 
-- **regex**: Matches the regex `/value.+/`.
-- **nregex**: Doesn't match the regex `/value.+/`.
+::: tip
+Visit [https://regex101.com/](https://regex101.com/) and select **Rust** syntax for assistance in creating expressions.
+:::
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:115]: Here we are missing the operators for Date/Time which are `gt` and `lt`

File: src/concepts/essentials/httpql.md [hunk 3]
```diff
@@ -2,99 +2,167 @@
...
+- `gt`: **Greater than** the supplied value.
+- `gte`: **Greater than or equal to** the supplied value.
+- `lt`: **Less than** the supplied value.
+- `lte`: **Less than or equal to** the supplied value.
+- `ne`: **Not equal** to the supplied value.
 
 ### String/Bytes
 
-This category of Operators works on Fields that are text values (or bytes) like `path` and `raw`.
+This category of Operators works on Fields that are text or byte values (_such as `path` and `raw`_).
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:85]: Do you want to bold Field same as you did in `integers`?

File: src/concepts/essentials/httpql.md [hunk 4]
```diff
@@ -2,99 +2,167 @@
...
+Autocomplete is not currently available when using the `source` Namespace.
+:::
+
+::: tip
+
+- When using the `source` Namespace - use all lowercase characters when naming the desired Caido feature.
+- If you are not receiving results of a `source` query - click the `Advanced` settings button next to the HTTPQL query bar to ensure the desired `Sources` are enabled.
+:::
+
+### roundtrip
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:146]: `date/time`

File: src/concepts/essentials/httpql.md [hunk 5]
```diff
@@ -2,99 +2,167 @@
...
-- **nregex**: Doesn't match the regex `/value.+/`.
+::: tip
+Visit [https://regex101.com/](https://regex101.com/) and select **Rust** syntax for assistance in creating expressions.
+:::
 
-## 4. Value
+## Value
 
-This is the value against which the Field will be compared. The value is either an integer (like `1`), a string (`"value"`) or a regex (`/value/`) depending on the Field and Operator.
+This is the value against which the Field will be compared. The value type is either an **integer**, a **string** or a **regex** depending on the Field and Operator.
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:118]: I would rephrase to something along: `The base types are integer, string, regex and date/time depending on the Field and/or Operator`

File: src/concepts/essentials/httpql.md [hunk 6]
```diff
@@ -2,99 +2,167 @@
...
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:119]: I think I would add the base types as h3 just so they are there?

File: src/concepts/essentials/httpql.md [hunk 7]
```diff
@@ -2,99 +2,167 @@
...
+- `workflow` - Traffic generated by a [Workflow](/reference/workflows/workflows.md).
+
+::: info
+Autocomplete is not currently available when using the `source` Namespace.
+:::
+
+::: tip
+
+- When using the `source` Namespace - use all lowercase characters when naming the desired Caido feature.
+- If you are not receiving results of a `source` query - click the `Advanced` settings button next to the HTTPQL query bar to ensure the desired `Sources` are enabled.
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:143]: Thats a good point, I would say the advanced is kinda of the same thing as the httpql syntax. We added the syntax for places where you dont have this advanced button like filter preset. So we should precise that tip is only valid for HTTP History and Search.

**Review by Sytten — CHANGES_REQUESTED** [review]

File: src/concepts/essentials/httpql.md (hunk 7 above)
Comment (ninjeeter) [reply src/concepts/essentials/httpql.md:143]: `source` is not available as a query in HTTP History unless I am mistaken.

File: src/concepts/essentials/httpql.md (hunk 4 above)
Comment (ninjeeter) [reply src/concepts/essentials/httpql.md:146]: This.

File: src/concepts/essentials/httpql.md [hunk 8]
```diff
@@ -2,99 +2,167 @@
...
 
 ### req
 
-- **ext**: The file extension (if we detected one). Extensions in Caido always contain the leading `.` (like `.js`). We are liberal in what we accept as an extension.
-- **host**: The hostname of the target server.
-- **method**: The HTTP Method used for the request in uppercase. If the request is malformed, this will contain the bytes read until the first space.
-- **path**: The path of the query, including the extension.
-- **port**: The port of the target server.
-- **raw**: The full raw data of the request. This allows you to search on things we currently don't index (like headers).
+- `ext`: The file extension (_if present_). Extensions in Caido always contain the leading `.` (_such as `.js`_).
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:42]: - `ext` (`String`)

File: src/concepts/essentials/httpql.md [hunk 9]
```diff
@@ -2,99 +2,167 @@
...
+- `host`: The hostname of the target server.
+- `method`: The HTTP Method used for the request in uppercase. If the request is malformed, this will contain the bytes read until the first whitespace.
+- `path`: The path of the query, including the extension.
+- `port`: The port of the target server.
+- `raw`: The full raw data of the request. This allows you to search by elements that Caido currently does not index (_such as headers_).
```
Comment (Sytten) [inline src/concepts/essentials/httpql.md:47]: - `raw` (`Bytes`)

**Review by Sytten — APPROVED** [review]
